### ISSUE TITLE
Error when running tests on latest devel

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#2277]
When running the tests on Mac (locally), the following error occurred (at random):

```
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Library/Python/2.7/site-packages/twisted/_threads/_threadworker.py", line 46, in work
    task()
  File "/Library/Python/2.7/site-packages/twisted/_threads/_team.py", line 190, in doWork
    task()
--- <exception caught here> ---
  File "/Library/Python/2.7/site-packages/twisted/python/threadpool.py", line 246, in inContext
    result = inContext.theWork()
  File "/Library/Python/2.7/site-packages/twisted/python/threadpool.py", line 262, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Library/Python/2.7/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Library/Python/2.7/site-packages/twisted/python/context.py", line 81, in callWithContext
    return func(*args,**kw)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/Modules/channel/channel.py", line 85, in _on_channel_created
    if channel_data[u'channel'].cid != self._channel_community.cid:
exceptions.AttributeError: 'NoneType' object has no attribute 'cid'
```

We've also seen this one in at least one other PR (#2274) during the test execution. I will fix it as separate commit in my PR #2253.

**Comment by lfdversluis** [comment]
This was introduced by @brussee and triggers at random. Also seen at https://jenkins.tribler.org/job/GH_Tribler_PR_tests_mac/1466/testReport/junit/Tribler.Test.Core.test_session/TestSessionAsServer/test_add_torrent_def_to_channel/ in #2265

**Comment by lfdversluis** [comment]
@brussee Any idea why this might be racy ?

**Comment by brussee** [comment]
@lfdversluis It sounds like a missing blocking_call_on_reactor_thread, but @devos50 made this:   File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/Modules/channel/channel.py", line 85, in _on_channel_created
@devos50 Do you know if that callback is supposed to be called on a certain thread or something along those lines?

**Comment by devos50** [comment]
@brussee I did not create that particular method.

I've found the problem and the fix is here (in my outstanding PR): https://github.com/Tribler/tribler/pull/2253/commits/6776fa74a120f148d1845edce178606db400eb8a. The fact that the notifier makes use of the thread pool by default can cause race conditions (in this case, the notification was fired after shutdown which is why `_channel_community` was `None`).
